### ISSUE TITLE
Error on analyzing gitlog

### ISSUE THREAD

**Issue by robinmuilwijk** [grimoirelab/GrimoireELK#14]
I have Perceval, Elastic and Kibana running (all tested).

I am following usecase https://github.com/grimoirelab/use_cases/blob/master/github/README.md. At the step where I clone my own repo, and than want to analyze it with the command:

`python3 ./p2o.py -e http://localhost:9200 --no_inc --debug git /tmp/bicho-gitlog.log`

I get the following errors (I am using the fosdem branch):

2016-06-19 14:13:08,296 Commit 29de50cf830480f0cbada8aed59d127d57ac9727 parsed
2016-06-19 14:13:08,296 Commit 1f9c503bb7e3009b7ce373d1f96c34768e9b7280 parsed
2016-06-19 14:13:08,297 Adding items to Ocean for <grimoire.ocean.git.GitOcean object at 0x7f00b25f9748> (100 items)
2016-06-19 14:13:08,297 Adding items to http://localhost:9200/git__tmp_ezplatform-gitlog.log/items/_bulk (in 100 packs)
2016-06-19 14:13:08,302 Error feeding ocean from git (/tmp/ezplatform-gitlog.log): 'commit'
Traceback (most recent call last):
  File "/home/osboxes/grimoire/GrimoireELK/utils/grimoire/arthur.py", line 76, in feed_backend
    ocean_backend.feed()
  File "/home/osboxes/grimoire/GrimoireELK/utils/grimoire/ocean/elastic.py", line 116, in feed
    self._items_to_es(items_pack)
  File "/home/osboxes/grimoire/GrimoireELK/utils/grimoire/ocean/elastic.py", line 144, in _items_to_es
    self.elastic.bulk_upload_sync(json_items, field_id)
  File "/home/osboxes/grimoire/GrimoireELK/utils/grimoire/elk/elastic.py", line 110, in bulk_upload_sync
    bulk_json += '{"index" : {"_id" : "%s" } }\n' % (item[field_id])
KeyError: 'commit'
2016-06-19 14:13:08,303 Adding repo to Ocean http://localhost:9200//conf/repos/git__tmp_ezplatform-gitlog.log {'backend_name': 'git', 'error': KeyError('commit',), 'backend_params': ['/tmp/ezplatform-gitlog.log'], 'success': False, 'repo_update': '2016-06-19T14:13:08.303627'}
Traceback (most recent call last):
  File "./p2o.py", line 99, in <module>
    args.backend, args.backend_args)
  File "/home/osboxes/grimoire/GrimoireELK/utils/grimoire/arthur.py", line 94, in feed_backend
    ConfOcean.add_repo(es_index, repo)
  File "/home/osboxes/grimoire/GrimoireELK/utils/grimoire/ocean/conf.py", line 67, in add_repo
    requests.post(url, data = json.dumps(repo))
  File "/usr/lib64/python3.4/json/__init__.py", line 230, in dumps
    return _default_encoder.encode(obj)
  File "/usr/lib64/python3.4/json/encoder.py", line 192, in encode
    chunks = self.iterencode(o, _one_shot=True)
  File "/usr/lib64/python3.4/json/encoder.py", line 250, in iterencode
    return _iterencode(o, 0)
  File "/usr/lib64/python3.4/json/encoder.py", line 173, in default
    raise TypeError(repr(o) + " is not JSON serializable")
TypeError: KeyError('commit',) is not JSON serializable
osboxes@osboxes:~/grimoire/GrimoireELK/utils>

**Comment by acs** [comment]
@robinmuilwijk fosdem branch is a bit outdated and we are not supporting it. Could you reproduce the same error in master branch?

**Comment by robinmuilwijk** [comment]
Jesus actually suggested using the fosdem branch, but I will try the master. I will report back with feedback.

**Comment by acs** [comment]
@robinmuilwijk yes, to follow the github use case use "fosdem" branch is the best approach, but it you find issues, we are not supporting this branch so probably, it is time to move to master (not sure in this scenario if the use case will work out of the box).

**Comment by jgbarah** [comment]
@acs I suggested to use the FOSDEM branch, because I'm not sure the github use case works with the current stack, as such. Well, I'm fact, I'm pretty much sure it doesn't. If you can just write here the sequence of commands, I can test it and update the case tu current Perceval and GrimoireELK.

@robinmuilwijk Could you please make sure you're using the FOSDEM version of Perceval too? The output format for Perceval changed after FOSDEM, and the kind of error you're seeing is very akin to others I've seen due to this change in format.

**Comment by robinmuilwijk** [comment]
I will switch to the FOSDEM branch of Perceval, thanks for the tip! I will post my feedback here, as soon as I've given it another run.

**Comment by jgbarah** [comment]
Great. Warning: maybe for Perceval it is a tag, not a branch, now I don't remember...

**Comment by robinmuilwijk** [comment]
Thanks for the heads up, I think it is a branch but I'll find out. Probably working on it tomorrow, setup Perceval again, and see if pushing the data to Elastic will work.

**Comment by robinmuilwijk** [comment]
Finally have some time to work on this. I moved to the fosdem16 branch for perceval, and the command I was stuck on now works. Continuing the GitHub usecase and see if I can finish the dashboard.

**Comment by jgbarah** [comment]
Great! Let us know if you have further problems. In any case, I hope to upgrade instructions to 0.3, which we're about to release, as soon as I can after the release. I'm closing this one for now, feel free to reopen / open a new issue if you find further problems.

**Comment by p23power** [comment]
Hi jgbarah,

Like robinmuilwijk, I am following usecase https://github.com/grimoirelab/use_cases/blob/master/github/README.md. .. I get a similar error at the following step:

`python3 ./p2o.py -e http://localhost:9200 --no_inc --debug git /tmp/bicho-gitlog.log`

using the fosdem branch:

`2016-08-29 22:10:35,401 Adding items to http://tiger-linux.domain.snsys.com:9200/git__tmp_bicho2-gitlog.log/items/_bulk (in 100 packs)
2016-08-29 22:10:35,407 Error feeding ocean from git (/tmp/bicho2-gitlog.log): 'hits' `

Using the master branch (or any of the other tags) I got a different issue i.e.  MySQLdb not found. 

`File "grimoire/GrimoireELK/utils/grimoire/elk/enrich.py", line 28, in <module>
import MySQLdb
ImportError: No module named 'MySQLdb' `

I have been googling all afternoon and cannot find a Python 3 compatible MySQLdb  (in fact everywhere says that  MySQLdb is not yet available for Python 3).

Please help.  Thanks

**Comment by robinmuilwijk** [comment]
@p23power I remember getting a warning/error on a missing Python dependency, for the database. I noticed it was not listed as requirement in the GitHub use-case, but went ahead and installed it, which let me continue the use-case. Not sure anymore which package I installed though, it could have been https://pypi.python.org/pypi/db.py.

**Comment by p23power** [comment]
Hi, 

My first error seems to be because of UTF-8 characters in the bicho git log.  I created a new log limiting the timeline to before feb 2016 and it worked ok.  
`2016-08-29 22:10:35,407 Error feeding ocean from git (/tmp/bicho2-gitlog.log): 'hits' `

I have progressed to attempting to mine a bugzilla - but this has auth and the fosdem branch version does not seem to use auth.  I took a look at the master branch, which seems to use auth, though I run back into the MySQLdb not found issue.  I installed the https://pypi.python.org/pypi/db.py and it didn't resolve this.

`File "grimoire/GrimoireELK/utils/grimoire/elk/enrich.py", line 28, in <module> import MySQLdb ImportError: No module named 'MySQLdb'`

Any more thoughts about this?

Many thanks
